presto-raptor/src/main/java/com/facebook/presto/raptor/storage/organization/ShardOrganizerUtil.java

```diff
@@ -116,14 +113,16 @@ private ShardOrganizerUtil() {}
                                 }
                             }
                             Optional<ShardRange> temporalRange = Optional.empty();
+                            Optional<Type> temporalType = Optional.empty();
                             if (temporalColumn.isPresent()) {
                                 temporalRange = getShardRange(ImmutableList.of(temporalColumn.get()), resultSet);
+                                temporalType = temporalColumn.map(TableColumn::getDataType);
```

Comment:
This should be moved below, so we don't have a `temporalType` without a `temporalRange`